xmodule/capa/capa_problem.py

```diff
@@ -495,6 +509,50 @@ def get_grade_from_current_answers(self, student_answers):
 
         return newcmap
 
+    def get_grade_from_answers(
+        self, student_answers: dict, correct_map: Optional[CorrectMap] = None
+    ) -> CorrectMap:
+        """
+        Gets the grade for the currently-saved problem state, but does not save it
+        to the block.
+
+        For new student_answers being graded, `student_answers` is a dict of all the
+        entries from request.POST, but with the first part of each key removed
+        (the string before the first "_").  Thus, for example,
+        input_ID123 -> ID123, and input_fromjs_ID123 -> fromjs_ID123.
+
+        For rescoring, `student_answers` is None.
+
+        If `correct_map` is provided, it is used as the correct_map for grading.
+
+        Calls the Response for each question in this problem, to do the actual grading.
```

Comment:
can we add here that this method is based on `get_grade_from_current_answers` but with X and Y modifications?